zookeeper-metrics-providers/zookeeper-prometheus-metrics/src/main/java/io/prometheus/client/SketchesSummary.java

```diff
@@ -0,0 +1,254 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.prometheus.client;
```

Comment:
Why do we have to locate `SketchesSummary` in this package?

Reply:
Since my intention here is not to implement a new type of Summary from scratch, but rather to extend the capabilities of SimpleCollector (such as metric definition, child management, and initialization) by inheriting from it. The `SketchesSummary.Builder` class needs access to some `package-private` fields in `io.prometheus.client.SimpleCollector.Builder` class, such as `labelNames` and `dontInitializeNoLabelsChild`.

Reply:
It would be nice if we don't need to put `SketchesSummary` in the io.prometheus.client package. 

Have you thought about submit the patch to the prometheus client library and have the issue fixed there?

Reply:
> Have you thought about submitting the patch to the Prometheus client library and fixing the issue there, so the fix will be leveraged by all users?

That's a great idea.

Since the sketch-based implementation introduces a delay in quantile values, it might not be acceptable for certain user scenarios. Btw, it's definitely worth discussing this approach with the maintainers of the Prometheus client library.